engines/bops_api/app/services/bops_api/application/parsers/address_parser.rb

```diff
@@ -11,15 +11,49 @@ def initialize(params)
         end
 
         def parse
-          {
-            uprn: params[:uprn],
-            address_1: "#{params[:pao]}, #{params[:street]}",
-            address_2: params[:organisation],
-            town: params[:town],
-            postcode: params[:postcode],
-            longitude: params[:longitude],
-            latitude: params[:latitude]
-          }
+          { uprn:, address_1:, address_2:, town:, postcode:, longitude:, latitude: }
+        end
+
+        private
+
+        def uprn
+          params[:uprn]
+        end
+
+        def sao
+          params.values_at(:sao, :saoEnd).compact_blank.join("-")
+        end
+
+        def pao
+          params.values_at(:pao, :paoEnd).compact_blank.join("-")
+        end
```

Comment:
Being pedantic, these should be en dashes, not hyphens. 🤓🧐

Reply:
Updated